azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/ResourceGroupApiMockTest.java

```diff
@@ -0,0 +1,154 @@
+   }
+
+   public void testListResourceGroupsReturns404() throws InterruptedException {
+      server.enqueue(response404());
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertTrue(isEmpty(resourceGroups));
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testListResourceGroupsWithOptions() throws InterruptedException {
+      server.enqueue(jsonResponse("/resourcegroups.json"));
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertEquals(size(resourceGroups), 2);
+      assertEquals(server.getRequestCount(), 1);
+
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testListResourceGroupsWithOptionsReturns404() throws InterruptedException {
+      server.enqueue(response404());
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertTrue(isEmpty(resourceGroups));
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testCreateResourceGroup() throws InterruptedException {
+      server.enqueue(jsonResponse("/resourcegroup.json").setStatus("HTTP/1.1 201 Created"));
+
+      HashMap<String, String> tags = new HashMap<String, String>();
+      tags.put("tagname1", "tagvalue1");
+
+      ResourceGroup resourceGroup = api.getResourceGroupApi(subscriptionid).create("jcloudstest", "West US", tags);
+
+      assertEquals(resourceGroup.name(), "jcloudstest");
+      assertEquals(resourceGroup.location(), "westus");
+      assertEquals(resourceGroup.tags().size(), 1);
+      assertTrue(resourceGroup.id().contains("jcloudstest"));
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "PUT", requestUrl + "/jcloudstest" + version, String.format("{\"location\":\"%s\", \"tags\":{\"tagname1\":\"tagvalue1\"}}", "West US"));
+   }
+
+   public void testGetResourceGroup() throws InterruptedException {
+      server.enqueue(jsonResponse("/resourcegroup.json"));
+
+      ResourceGroup resourceGroup = api.getResourceGroupApi(subscriptionid).get("jcloudstest");
+
+      assertEquals(resourceGroup.name(), "jcloudstest");
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", requestUrl + "/jcloudstest" + version);
+   }
+
+   public void testGetResourceGroupReturns404() throws InterruptedException {
+      server.enqueue(response404());
+
+      ResourceGroup resourceGroup = api.getResourceGroupApi(subscriptionid).get("jcloudstest");
+
+      assertNull(resourceGroup);
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", requestUrl + "/jcloudstest" + version);
+   }
+
+   // Commenting this out due to this error during CI run: Command not considered safe to retry because request method is PATCH
```

Comment:
Thanks for the feedback @nacx! Question, how come the mock tests and live tests all work for PATCH when I run this locally when the default java driver does not support PATCH? What is a good way to test this locally to ensure this is really using OkHttp instead of the default java driver?